lib/sla.js

```diff
@@ -84,12 +123,38 @@ function createChannel(client, bridge) {
  * @param {Object} client - Client received from app.js.
  * @return {Q} Q promise object.
  */
-module.exports = function(client) {
-  return designateMixingBridge(client)
-    .then(function (bridge) {
-      return createChannel(client, bridge)
-        .then(function (channel) {
-          return addChannelsToBridge(channel, bridge);
-        });
-    });
+module.exports = function(client, channel, bridgeNumber) {
+  //This specifies what bridge number to use for SLA
+  if(!isNaN(parseInt(bridgeNumber))) {
+    var answer = Q.denodeify(channel.answer.bind(channel));
+    answer();
+    return designateMixingBridge(client, bridgeNumber)
+      .then(function (bridge) {
+        return createChannel(client, channel, bridge)
+          .then(function (dialed) {
+            if (dialed) {
+              return addChannelsToBridge(channel, dialed, bridge);
+            }
+            else {
+              var hangup = Q.denodeify(channel.hangup.bind(channel));
+              hangup(function(channel) {
+                 console.log('Hanging up inbound channel %s', channel.id);
+              });
+              var defer = Q.defer();
+              defer.resolve('Channel rejected or caller hung up');
+              return defer.promise;
```

Comment:
You are already inside a promise chain, you can just return the string 'Channel rejected or caller hung up' if that is the result you want resolved.